### PR TITLE
Dynamically getting the JSPS version.

### PR DESCRIPTION
**Pull request by safacero**
Issue #
When we migrate a project we used a hardcoded version of JSPS. I updated that version as a backup but also it's better to dynamically pick it up from the version that VS installs.

### CODE REVIEW

File: Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs [hunk 1]
```diff
@@ -110,9 +114,9 @@ private static JObject ReadPackageJson(string projectDir)
 
         private static string GetSdkVersion()
         {
-            // TODO: figure out if sdk is installed on machine?
-            // do i need to do that or will VS automatically install given version
-            throw new NotImplementedException();
+            // Use the installed version of JSPS on the NuGet fallback folder.
+            var versions = Directory.GetDirectories(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), "Microsoft Visual Studio", "Shared", "NuGetPackages", "microsoft.visualstudio.javascript.sdk"));
+            return Path.GetFileName(versions.OrderByDescending(v => v).FirstOrDefault());
```
Comment (PranavSenthilnathan) [inline Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs:119]: I think this does lexicographic sorting so version 2 would be before version 10.

File: Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs [hunk 2]
```diff
@@ -110,9 +114,14 @@ private static JObject ReadPackageJson(string projectDir)
 
         private static string GetSdkVersion()
         {
-            // TODO: figure out if sdk is installed on machine?
-            // do i need to do that or will VS automatically install given version
-            throw new NotImplementedException();
+            // Use the installed version of JSPS on the NuGet fallback folder.
+            var versions = Directory.GetDirectories(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), "Microsoft Visual Studio", "Shared", "NuGetPackages", "microsoft.visualstudio.javascript.sdk"));
+            var newestVersion = versions
+                .Select(v => new { Version = new Version(Path.GetFileName(v)) })
```
Comment (PranavSenthilnathan) [inline Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs:120]: This should have some error handling (`Version.TryParse`). Nuget packages can have suffixes (our JSPS package used to have the `-alpha` suffix) which I think would throw an exception here.

Comment (safacero) [reply Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs]: Yeah it did! Fixed it now. I don't think we're using that anymore, right?

File: Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs [hunk 3]
```diff
@@ -110,9 +114,19 @@ private static JObject ReadPackageJson(string projectDir)
...
-            // TODO: figure out if sdk is installed on machine?
-            // do i need to do that or will VS automatically install given version
-            throw new NotImplementedException();
+            // Use the installed version of JSPS on the NuGet fallback folder.
+            var versions = Directory.GetDirectories(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), "Microsoft Visual Studio", "Shared", "NuGetPackages", "microsoft.visualstudio.javascript.sdk"));
+            var newestVersion = versions
+                .Select(v =>
+                {
+                    Version version;
+                    Version.TryParse(Path.GetFileName(v), out version);
```
Comment (PranavSenthilnathan) [inline Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs:123]: `out var version`

File: Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs [hunk 4]
```diff
@@ -43,6 +40,13 @@ public static string WriteEsproj(string projectDir, string projectName, string s
             }
 
             EsprojFile esprojFile = new EsprojFile() { PropertyGroup = propGroup };
+            var sdkVersion = GetSdkVersion();
+
+            // If the SDK version is not found on NuGet fallback folder, keep the one hardcoded on EsProjFileModel
+            if (sdkVersion != null)
+            {
+                esprojFile.Sdk = $"Microsoft.VisualStudio.JavaScript.Sdk/{GetSdkVersion()}";
```
Comment (PranavSenthilnathan) [inline Nodejs/MigrateToJsps/Jsps/EsprojFileWriter.cs:48]:
```suggestion
                esprojFile.Sdk = $"Microsoft.VisualStudio.JavaScript.Sdk/{sdkVersion}";
```

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**